### DISCUSSION TITLE
My server got spammed and destroyed 2 calves

### DISCUSSION THREAD

**Discussion by ghost** [Pryaxis/TShock#1708] [General]
Last night I was hacked into a server and spamed a lot of strange characters with numbers, and then he threw away a lot of stuff.Is there any way to prevent spam and fix the 2-wall error when going into causing game crashes
![Screenshot_2020-04-25-07-23-17-573_com discord](https://user-images.githubusercontent.com/46393687/80266492-6dae2f80-86c6-11ea-8c22-8dddeb268fdd.jpg)
![Screenshot_20200425072132](https://user-images.githubusercontent.com/46393687/80266525-8e768500-86c6-11ea-8cd7-358290f11a28.jpg)

**Comment by moisterrific** [comment]
Which version of TShock are you running?

**Comment by hakusaro** [comment]
At least in terms of your game world, you should open the world in `TEdit` and save it off. In general that's a good practice as it quite easily deals with anything that could be corrupted on the world itself or doesn't play nicely with Terraria anymore. Obviously, make a backup before you do this.

**Comment by sors89** [comment]
If you want to prevent player from spamming, just use AntiSpam by MarioE
[https://github.com/Exodius-GG/AntiSpam/raw/master/AntiSpam/bin/AntiSpam.dll](url)

**Reply by moisterrific** [reply]
I’ve tried that, has way too many false positives. When you open many treasure bags or crates in quick succession it gets detected as spam lol

**Reply by sors89** [reply]
Oh, seems like i'm need to fix some stuff in it lmao

**Reply by ghost** [reply]
I run in tshock mobile and erro

**Reply by ghost** [reply]
![Screenshot_2020-04-25-11-55-26-784_com microsoft rdc android](https://user-images.githubusercontent.com/46393687/80284775-ac85c900-874a-11ea-8774-0527d24045f5.jpg)

**Reply by moisterrific** [reply]
TShock mobile forks are not officially supported. The error messages you’re seeing are because of the plugin, not TShock. Likely the plugin wasn’t properly ported to be compatible with the mobile version.

**Reply by moisterrific** [reply]
It says method not found because the hook is trying to reference code that do not exist in the unofficial mobile port. You can easily fix it by checking what the method is for whichever mobile fork you’re using and updating it for compatibility.

**Reply by ghost** [reply]
How to search method

**Reply by sors89** [reply]
Can i get your references in your ServerPlugins folder @NOTAD ?

**Reply by ghost** [reply]
Oke

**Reply by ghost** [reply]
http://f.notad.net/f/191622867abd45e9adce/

**Comment by bartico6** [comment]
There's a bunch of things you can do to prevent spam, but all of the solutions come at a cost to players.  
You can throttle chat by imposing a limit on how often you can post messages, but bots who rapidly join and leave will get around that.  
You can disable chatting for the first N seconds of your connection, but players will have to wait to say hello and they'll complain.    
You can try to ban abusive IPs using some sort of blacklist, but you'll falsely flag people fairly often because dynamic IPs exist.  
You can try to throttle login rates, but that'll make your server unreachable during bot attacks.  

Generally speaking, there isn't much you **can** do without affecting legitimate players. Welcome to Terraria.
